qos-service-impl/src/main/java/com/palantir/atlasdb/qos/ratelimit/guava/RateLimiter.java

```diff
@@ -235,11 +237,11 @@ public final void setRate(double permitsPerSecond) {
         checkArgument(
                 permitsPerSecond > 0.0 && !Double.isNaN(permitsPerSecond), "rate must be positive");
         synchronized (mutex()) {
-            doSetRate(permitsPerSecond, stopwatch.readMicros());
+            doSetRate(permitsPerSecond, stopwatch.readNanos());
```

Comment:
why the change from micros to nanos? it's generally nicer, but without a strong need we should prefer avoiding changes to this copied code

Reply:
If you have longs you might often need to wait via sleeping for just a few nanos, as sleeping in micros would be excessive.

Reply:
If you have longs you might often need to be sleeping for just a few nanos, as the `stableand sleeping in micros would be non-performant.